### PR TITLE
Mini Protocol Mux Limts

### PR DESCRIPTION
**Pull request by coot**
This PR consists of two major parts: estimations of inbound queue size for each mini-protocol, plus a refactoring to gather parameters that are needed to set them in one central place.

* calculations of limits for each protocol (every one in each commit)
* when we have those we can add 'MiniProtocolParameters' which take all the required data to compute the limits: chain sync pipelineing parameters, block fetch pipelineing & tx submission unacknowledged `TxIds` (from which one can compute upper bound for pipelining).

`MiniProtocolParameters` are set by `mkNodeArgs` function in `ouroboros-consensus` and passed down the stack.  `NodeArgs` can be change by the `cardano-node`, this makes us safe from changing the parameters at some level without making required changes in `ouroboros-network` - which potentially could end-up badly.

### CODE REVIEW

**Comment by coot** [comment]
This PR addresses #575.

File: ouroboros-network/src/Ouroboros/Network/NodeToNode.hs [hunk 1]
```diff
@@ -160,6 +161,75 @@ nodeToNodeProtocols chainSync
...
       }
     ]
+  where
+    -- TODO: these limits depends on various parameters instantiated by
+    -- 'Ouroboros.Consensus.Node'.  We should parametrize over them, but these
+    -- limits must only increase.
+    chainSyncProtocolLimits :: MiniProtocolLimits
+    chainSyncProtocolLimits =
+      MiniProtocolLimits {
+          -- chain sync has two potencially large messages:
```
Comment (njd42) [inline ouroboros-network/src/Ouroboros/Network/NodeToNode.hs]: potencially  -> potentially

File: ouroboros-network/src/Ouroboros/Network/NodeToNode.hs [hunk 2]
```diff
@@ -165,7 +165,8 @@ nodeToNodeProtocols chainSync
     -- TODO: these limits depends on various parameters instantiated by
     -- 'Ouroboros.Consensus.Node'.  We should parametrize over them, but these
```
Comment (njd42) [inline ouroboros-network/src/Ouroboros/Network/NodeToNode.hs]: parametrize -> parameterize (or -....se - I'm UK,...)

File: ouroboros-network/src/Ouroboros/Network/NodeToNode.hs [hunk 3]
```diff
@@ -252,19 +292,77 @@ nodeToNodeProtocols chainSync
...
+          -- the 'TxId' without the size of 'Tx' in bytes).  So the ingress
+          -- queue of 'txSubmissionOutbound' is bounded by the ingress side of
+          -- the 'txSubmissionInbound'
+          --
+          -- Currently the value of 'txSubmissionMaxUnacked' is '100', for
+          -- which the upper bound is `100 * (44 + 65_540) = 6_558_400`, we add
+          -- 10% as a safety margin.
+          --
+          maximumIngressQueue = addSafetyMargin $
+              fromIntegral txSubmissionMaxUnacked * (44 + 65540)
```
Comment (njd42) [inline ouroboros-network/src/Ouroboros/Network/NodeToNode.hs]: for consistency (given you've turned on the feature) you could make it `(44 + 65_540)`

**Comment by coot** [comment]
>  I did get my calculator out to check no schoolboy arithmetical errors - I found none.
Thanks for doing that!

I managed to sync the node from mainnet with this patch without problems.

File: ouroboros-network/src/Ouroboros/Network/NodeToNode.hs (hunk 2 above)
Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/NodeToNode.hs]: o really, in UK its spelled this way, I wouldn't think.

Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/NodeToNode.hs]: This comment needs to be removed.  I fixed it in a later patch.

**Comment by coot** [comment]
@karknu I created an issue for that: #1802

**Pull request merged**